base/range.jl

```diff
@@ -182,33 +183,33 @@ end
 
 function linspace{T<:AbstractFloat}(start::T, stop::T, len::T)
```

Comment:
First of all, thanks for taking a look at this!

Certainly the length of a `LinSpace` should not have units. But from the user's perspective, the idiomatic way to get the length of a `LinSpace` is to call `length`, not to access the `len` field. It is easy to have `length` return a number without units.

Perhaps more importantly, I didn't want to put assumptions I've made in my Unitful package into base. The way I've done this now, the numeric types are still forced to be the same, consistent with prior logic (`start`, `stop`, and `len` are all of type `T`). Regardless of how one implements numbers with units (I call them `Quantity`s), it's pretty clear that they would use parametric types to allow for different numeric types (`Float32` vs `Float64` or whatever), and since type parameters are invariant, the numeric types must always be the same, just like they are now. Does that clarify my reasoning?